File: leetcode-algorithms/066. Plus One/PlusOne.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        int i=digits.Length-1;
        digits[i]++;
        while(digits[i]==10&&i>0){
            digits[i]=0;
            digits[i-1]++;
            i--;
        }
        if(digits[0]==10){
            int[] newDigits = new int[digits.Length+1];
            newDigits[0]=1;
            return newDigits;
        }else
            return digits;
    }
}

File: leetcode-algorithms/066. Plus One/PlusOne.py
class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        length = len(digits)
        while(length != 0 and digits[length-1] == 9):
            digits[length-1] = 0
            length -= 1
        if(length == 0):
            digits.insert(0,1)
        else:
            digits[length-1] += 1
        return digits